spec/integration/resolve_reference_spec.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+require 'bolt_spec/run'
+require 'open3'
+
+describe 'terraform::resolve_reference' do
+  # include BoltSpec::Config
+  # include BoltSpec::Integration
+  include BoltSpec::Run
+
+  let(:bolt_config) { { 'modulepath' => RSpec.configuration.module_path } }
+  let(:terraform_dir) { File.join(RSpec.configuration.module_path, '../docker_provision') }
+
+  let(:resource_type) { 'docker_container' }
+  let(:target_mapping) { { 'uri' => 'ip_address' } }
+  let(:expected_result) { { 'value' => [{ 'uri' => '172.17.0.2' }] } }
+
+  before(:all) do
+    terraform_dir = File.join(RSpec.configuration.module_path, '../docker_provision')
+    _out, _err, status = Open3.capture3('terraform init', chdir: terraform_dir)
+    expect(status).to eq(0)
+    _out, _err, status = Open3.capture3('terraform apply -auto-approve', chdir: terraform_dir)
+    expect(status).to eq(0)
+  end
+
+  after(:all) do
+    terraform_dir = File.join(RSpec.configuration.module_path, '../docker_provision')
+    _out, _err, status = Open3.capture3('terraform destroy -auto-approve', chdir: terraform_dir)
+    expect(status).to eq(0)
+  end
+
+  it 'resolves references from an applied terraform manifest' do
+    params = {
+      'dir' => terraform_dir,
+      'resource_type' => resource_type,
+      'target_mapping' => target_mapping
+    }
+
+    result = run_task('terraform::resolve_reference', 'localhost', params)
```

Comment:
Others may feel this is overkill, but I would love to see a test where we lay down an inventoryfile  that uses the terraform plugin and actually run a command on a running container that was dynamically discovered.